tests/integration/UploadSlideBackTranslationIntegrationTest.php

```diff
@@ -0,0 +1,317 @@
+        $this->assertEquals(file_get_contents($storyFile), base64_decode($payload['Data']));
+    }
+
+
+    /**
+     * @depends testCreateNewStory
+     *
+     * @return void
+     * @throws GuzzleException
+     * @throws \Exception
+     */
+    public function testUpdateExistingStory()
+    {
+        // payload for creating a new story
+        $payload = [
+            'Key' => '',
+            'PhoneId' => 'lmnopq',
+            'Data' => base64_encode('Test data for story file'),
+            'TemplateTitle' => $this->storyTemplate,
+            'IsWholeStory' => "true",
+            'Language' => 'es',
+        ];
+
+        $response = $this->httpClient->request("POST", self::$host . $this->uri, ['form_params' => $payload]);
+        $responseJson = $response->getBody()->getContents();
+        $this->assertEquals(200, $response->getStatusCode());
+
+        $responseArr = json_decode($responseJson, true);
+        $storyId = $responseArr['StoryId'];
+
+        # verify story is created correctly in the database
+        $stories = $this->getStories($storyId);
+        $this->assertCount(1, $stories);
+        $story = $stories[0];
+
+        $this->assertEquals($payload['Language'], $story['language']);
+        $this->assertEquals($payload['TemplateTitle'], $story['title']);
+        $this->assertEquals($this->getProjectId($payload['PhoneId']), $story['projectId']);
+
+
+        # verify slides are all created correctly in the database
+        $slides = $this->getSlides($storyId);
+        $this->assertCount(3, $slides);
+
+
+        $storyFile = sprintf("%s/%s/%s/wholeStory.m4a", self::$uploadedProjectDir, $payload['PhoneId'], $storyId);
+        $this->assertFileExists($storyFile);
+        $this->assertEquals(file_get_contents($storyFile), base64_decode($payload['Data']));
+
+
+        // Update story created above identified by $storyId
+
+        // payload for updating a new story
+        $payload = [
+            'Key' => '',
+            'PhoneId' => 'lmnopq',
+            'Data' => base64_encode('Story data has been updated'),
+            'TemplateTitle' => 'Template title updated',
+            'IsWholeStory' => "true",
+            'Language' => 'pl',
+            'StoryId' => $storyId
+        ];
+
+        $response = $this->httpClient->request("POST", self::$host . $this->uri, ['form_params' => $payload]);
+        $responseJson = $response->getBody()->getContents();
+        $this->assertEquals(200, $response->getStatusCode());
+
+        $responseArr = json_decode($responseJson, true);
+
+        // should not create a new story
+        $this->assertEquals($storyId, $responseArr['StoryId']);
+
+        # verify story is updated correctly in the database
+        $stories = $this->getStories($storyId);
+        $this->assertCount(1, $stories);
+        $story = $stories[0];
+
+        // assert that story data isn't impacted with story update request
+        $this->assertEquals('es', $story['language']);
```

Comment:
Can you name the payload differently, e.g. $createStoryPayload and $updateStoryPayload so that you can use it in the assertions? That way, it's super clear to readers that we are expecting $story to be the same as $createStoryPayload and not $updateStoryPayload.

Reply:
Can you update the tests so that it's clearer that it's the create story values and not just arbitrary strings? e.g. `$this->assertEquals($createStoryPayload['Language'], $story['language']);`
Here and below.